src/main/java/com/kelinci/mainapp/api/RegisteredUserMapper.java

```diff
@@ -0,0 +1,22 @@
+package com.kelinci.mainapp.api;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+public class RegisteredUserMapper {
+    public RegisteredUserMapper() {
+    }
+
+    public List<RegisteredUserResponse> getListOfRegisteredUsersEmails() {
+        SimpleUserDatabase userDatabaseController = new SimpleUserDatabase();
```

Comment:
Unikamy jak ognia tworzenia obiektów, które coś robią - w metodach. Dlaczego? Dlatego, że jak ta metoda zostanie wywołana 1 000 000 razy, to Ty utworzysz milion obiektów klasy SimpleUserDatabase, które żyć będą tylko do linii 16.

Rozwiązaniem ogólnym jest wrzucenie tego obiektu jako pola klasy. Bo i tak w 100% przypadków będziemy chcieli, byśmy mieli JEDEN obiekt na CAŁĄ aplikację, który coś wykonuje (mappuje -> Mapper, wyciąga coś z bazy -> entityManager, itp itd)

Natomiast tutaj mamy głupiutkiego mappera, który powinien wymusić żeby zapodać mu to co potrzebuje do mapowania (Listę OurUserów) jako parametr metody. I wtedy on sobie to przemapuje na List<RegisteredUserResponse>

Także prośba, by mapper nie umiał wyciągać sobie listę our userów z bazy danych ;)

Reply:
ok,  zrobione